### ISSUE TITLE
Modbus RTU Client on ESP32

### ISSUE THREAD

**Issue by maweicoltsmart** [zephyrproject-rtos/zephyr#36575]
**Describe the bug**
Make the following modifications to the esp32.dts file:
delete file zephyr\samples\subsys\modbus\rtu_client\app.overlay
```
&uart2 {
	status = "okay";
	current-speed = <115200>;
	tx-pin = <17>;
	rx-pin = <16>;
	/* rts-pin = <11>; */
	/* cts-pin = <6>; */
	/* hw-flow-control; */
	modbus0 {
		compatible = "zephyr,modbus-serial";
		label = "MODBUS0";
		status = "okay";
		de-gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;	/* D9 */
	};
};
```
make and flash：west build -b esp32 samples\subsys\modbus\rtu_client

After the program runs, Frequently call cb_handler_tx(ctx);.
File: modbus_serial.c
Function: uart_cb_handler
Endless loop: 

```
if (uart_irq_tx_ready(cfg->dev)) {
cb_handler_tx(ctx);
}

```

**Comment by sylvioalves** [comment]
@maweicoltsmart Can you inform which esp32 board you have been working with?

**Comment by ramb0t** [comment]
I'm still getting to the bottom of this issue, but I have found there are some weaknesses in both the ESP32 uart driver implementation (#41526) and in the general modbus subsystem implementation (see the fix here #40578) 
For anyone with the same problem please see my commits in this branch https://github.com/ramb0t/zephyr/tree/esp32_modbus_driver_fix
This is currently working for my ESP32 modbus system, but I think either the ESP32 uart driver or the modbus_serial uart_cb_handler() while loop needs some improvement to prevent infinite loops
